com.unity.render-pipelines.core/Runtime/Volume/VolumeParameter.cs

```diff
@@ -1641,14 +1642,239 @@ internal override void Interp(VolumeParameter from, VolumeParameter to, float t)
+
+            dstValue = (t * (t * (t * c0 + c1) + c2)) + c3;
+            dstDeriv = (t * (3.0f * t * c0 + 2.0f * c1)) + c2;
+        }
+
+        // lhsIndex and rhsIndex are the indices in the keys array. The lhsIndex/rhsIndex may be -1, in which it creates a synthetic first key
+        // at startTime, or beyond the length of the array, in which case it creates a synthetic key at endTime.
+        static private Keyframe EvalKeyAtTime(Keyframe[] keys, int lhsIndex, int rhsIndex, float startTime, float endTime, float currTime)
+        {
+            Assert.IsTrue(keys.Length >= 1);
+            Assert.IsTrue(startTime <= keys[0].time);
+            Assert.IsTrue(keys[keys.Length - 1].time <= endTime);
+
+            Keyframe lhsKey = FetchKeyFromIndexClamped(keys, lhsIndex, startTime, endTime);
+            Keyframe rhsKey = FetchKeyFromIndexClamped(keys, rhsIndex, startTime, endTime);
+
+            float currValue = 0.0f;
+            float currDeriv = 0.0f;
+            EvalCurveSegmentAndDeriv(ref currValue, ref currDeriv, lhsKey, rhsKey, currTime);
+
+            Keyframe dstKey = new Keyframe(currTime, currValue, currDeriv, currDeriv);
+            return dstKey;
+        }
+
+        public override void Interp(AnimationCurve lhsCurve, AnimationCurve rhsCurve, float t)
+        {
+            if (t <= 0.0f)
+            {
+                m_Value = lhsCurve;
+            }
+            else if (t >= 1.0f)
+            {
+                m_Value = rhsCurve;
+            }
+            else
+            {
+                m_Value = new AnimationCurve();
+
+                Assert.IsTrue(lhsCurve.keys.Length >= 1);
+                Assert.IsTrue(rhsCurve.keys.Length >= 1);
+
+                // first, figure out the start and end time to include both curves
+                float startTime = Mathf.Min(lhsCurve.keys[0].time, rhsCurve.keys[0].time);
+                float endTime = Mathf.Max(lhsCurve.keys[lhsCurve.length - 1].time, rhsCurve.keys[rhsCurve.length - 1].time);
+
+                // we don't know how many keys the resulting curve will have (because we will compact keys that are at the exact
+                // same time), but in most cases we will need the worst case number of keys. So allocate the worst case, and resize
+                // the array at the end if we need to compact.
+                int maxNumKeys = lhsCurve.length + rhsCurve.length;
+                int currNumKeys = 0;
+                Keyframe[] dstKeys = new Keyframe[maxNumKeys];
+
+                int lhsKeyCurr = 0;
+                int rhsKeyCurr = 0;
+
+                while (lhsKeyCurr < lhsCurve.keys.Length && rhsKeyCurr < rhsCurve.keys.Length)
+                {
+                    // the index is considered invalid once it goes off the end of the array
+                    bool lhsValid = lhsKeyCurr < lhsCurve.keys.Length;
+                    bool rhsValid = rhsKeyCurr < rhsCurve.keys.Length;
```

Comment:
I think these two variables are always true given the end condition of the loop. Code below can be simplified

Reply:
Hrm, good catch. That actually looks like a bug. It should be OR, not AND. I'll dig into it and update code.

Reply:
instead of duplicating the condition check, try perhaps moving this outside of the loop and update it at the end of the loop. Here, first time we do this check two time for no reason.